pulsar-functions/utils/src/main/java/org/apache/pulsar/functions/utils/Utils.java

```diff
@@ -130,7 +131,12 @@ public static int findAvailablePort() {
             if (userClass instanceof Function) {
                 Function pulsarFunction = (Function) userClass;
                 typeArgs = TypeResolver.resolveRawArguments(Function.class, pulsarFunction.getClass());
-            } else {
+            }
+            else if (userClass instanceof Consumer){
```

Comment:
Is this code tested?  I don't think this currently is even going to run.  Many places in the functions code we assume there are two types returned by the this function.

Reply:
Not tested, if we can agree on the approach I will add the remaining code and tests.